src/messaging.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * collection of functions to show messages with actions in a consistent manner
+ */
+
+import * as vscode from 'vscode'
+
+export interface MessageAction {
+  title: string
+  action: () => void
+}
+
+export function systemErrorMessage(message: string, ...actions: Array<MessageAction>) {
+  const msg = vscode.window.showErrorMessage(message, ..._extractActionTitles(actions))
+  if (msg) {
+    msg.then(_handleMessageActions(actions))
+  }
+}
+
+export function systemWarningMessage(message: string, ...actions: Array<MessageAction>) {
+  const msg = vscode.window.showWarningMessage(message, ..._extractActionTitles(actions))
+  if (msg) {
+    msg.then(_handleMessageActions(actions))
+  }
+}
+
+// common actions
+export const showTroubleshootingAction: MessageAction = {
+  title: 'help',
+  action: () => vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(TroubleShootingURL)),
+}
+
+export const TroubleShootingURL = 'https://github.com/jest-community/vscode-jest/blob/master/README.md#troubleshooting'
+//
+// internal methods
+//
+function _extractActionTitles(actions?: Array<MessageAction>): string[] {
+  if (!actions || actions.length === 0) {
```

Comment:
`map` also works with an empty array, the `!actions` check on its one should therefore be enough.